src/test/java/com/squareup/javapoet/ParameterSpecTest.java

```diff
@@ -60,6 +60,25 @@ private TypeElement getElement(Class<?> clazz) {
     assertThat(a.toString()).isEqualTo(b.toString());
   }
 
+  @Test public void receiverParameterInstanceMethod() {
+    ParameterSpec.Builder builder = ParameterSpec.builder(int.class, "this");
+    assertThat(builder.build().name).isEqualTo("this");
+  }
+
+  @Test public void receiverParameterNestedClass() {
+    ParameterSpec.Builder builder = ParameterSpec.builder(int.class, "Foo.this");
```

Comment:
Oh crap this is scary. Does this work with `Foo.Bar.this`? Or `com.example.Foo.this`? If so, that means it probably needs to participate in the automatic import system to ensure you're referring to what you actually mean.

Reply:
Luckily `Foo.this` can only be used from the constructor of a direct nested type of `Foo`, so no import required.

In fact, the grammar only allows a single unqualified identifier, so even if you managed to make it ambiguous (another inner nested type with the outer types name?), there's nothing JavaPoet could do to make it compile I think.

```
ReceiverParameter:
{Annotation} UnannType [Identifier .] this
```

That said, this is the simplest solution but probably not the most elegant - if the grammar was more complicated, I might suggest `MethodSpec#receives(TypeName)` with the identifier derived automatically during emission instead.

Reply:
I'm not worried about it then. Thanks! Frankly I don't expect many to ever need this feature as-is but it's so easy to support we might as well add it.

Reply:
I agree - relevant receiver annotations seem pretty rare (I only know the one), and needing one in a generated class even more so.

If support for Java 8 annotations was otherwise lacking I'd probably have just hacked around it, but it seems very thorough. Thanks for looking!